### ISSUE TITLE
Adding admonitions

### ISSUE THREAD

**Issue by Fabrizz** [chrismwilliams/astro-theme-cactus#285]
Hey I was thinking in adding admonition support to this astro theme, the question is how would you like the markdown to be? Some projects like using
````mdx
<tip>This is a tip<tip/>

<admonition type="tip">This is a tip<admonition/>

```ad-tip
This is a tip
```

````

There are some packages that will do this but customizing the output to add for example the admonition icon like Github does:
> [!TIP]
> Optional information to help a user be more successful.

> [!WARNING]  
> Critical content demanding immediate user attention due to potential risks.

Here I done it for https://github.com/saicaca/fuwari/pull/77, I discovered this project as I was doing that PR.

**Comment by chrismwilliams** [comment]
Hi @Fabrizz,

thanks for opening this as I was going to do this a while ago but forgot to pin it on my board. I think the best option would be to follow Astro's example in [Starlight](https://starlight.astro.build/guides/components/#asides), basically your second example, as users may be more familiar with the syntax.

Let me know if you need any help.

**Comment by Fabrizz** [comment]
Great, I'm going to write it as an internal Rehype component. It could be a mdx component tho

**Comment by chrismwilliams** [comment]
Yeah a rehype plugin would be great. Not sure why they went the component route as it will be `.mdx` only

**Comment by chrismwilliams** [comment]
Closed #291
